src/app/shared/services/file-manager/file-manager.service.ts

```diff
@@ -75,4 +76,20 @@ export class FileManagerService extends SyncServiceBase {
       encoding: Encoding.UTF8,
     });
   }
+
+  /** @returns filepath and file size metadata from a file saved in local storage */
+  async generateAssetContentsEntry(relativePath: string): Promise<Partial<IAssetEntry>> {
```

Comment:
nit(non-blocking)
Shouldn't all the metadata be available from the asset pack to begin with? Do we need to be generating (particularly if it means no option for an md5 hash)